### ISSUE TITLE
Create FME Service does not properly encode username and password

### ISSUE THREAD

**Issue by RichieBRU** [vertigis/workflow-activities-fme#12]
As below I seem to be getting a valid service object but am unable to get workspace parameters.

Plugging that parameters URL into my browser and authenticating with the same account resolves the parameters json ok.

![image](https://user-images.githubusercontent.com/87160657/125025383-ecbe3680-e0c5-11eb-9abd-a00e96f39e78.png)

**Comment by rcooney** [comment]
@RichieBRU is there an `Authorization` header on the failed request to get the parameters?

**Comment by RichieBRU** [comment]
Authorization:
	fmetoken token=Both user ID and password need to be specified for viewing/generating token.

**Comment by rcooney** [comment]
@RichieBRU it appears to be an encoding problem. The FME API (that is embedded in the activity pack) doesn't properly URL encode the username and password parameters. https://github.com/geocortex/workflow-activities-fme/blob/d8e9f990e0bcaa203105baabc2498549d5045cd4/src/FMEServer.js#L979

You could confirm this and work around it in the workflow by setting the password input with an expression like this `=encodeURIComponent("your-password")`

**Comment by RichieBRU** [comment]
Thanks @rcooney 

Parameter array is now being returned however the next problem is the run fme job. Not sure what expected scalar means below in the output of that activity.
![image](https://user-images.githubusercontent.com/87160657/126414364-a2d3b505-c6de-4d40-9781-e1a52babd571.png)

**Comment by rcooney** [comment]
@RichieBRU it looks like you are passing the `result` parameters array from the **Get FME Workspace Parameters** activity directly into the Parameters input of **Run FME Job** activity. 

The Parameters input of **Run FME Job** activity is expecting an object of keys and values that might look something like this:
```
={
  Username: "your-name",
  foo: "bar",
  ParameterX: "some-value",
  ParameterY: 2,
  ParameterZ: true
}
```

You'll have to take the `name`s from the **Get FME Workspace Parameters** results and use it to build this object.

**Comment by rcooney** [comment]
I'll make the code change to automatically call `encodeURIComponent` on the username/password inputs. This is going to be a breaking change though. You'll have to remove your workaround once it is released as `v2.0.0`.
